samples/nrf9160/modem_shell/src/drivers/nrf91_non_offloaded_dev.c

```diff
@@ -0,0 +1,495 @@
+			(void)inet_pton(AF_INET6, dns_addr_str, addr6);
+		}
+
+		/* Read link MTU if exists: */
+		if (iterator == 1) {
+			ret = at_params_int_get(&param_list,
+						AT_CMD_PDP_CONTEXT_READ_DYN_INFO_MTU_INDEX,
+						&(ret_dyn_info->ipv6_mtu));
+			if (ret) {
+				/* Don't care if it fails: */
+				ret = 0;
+				ret_dyn_info->ipv6_mtu = 0;
+			}
+		} else {
+			ret = at_params_int_get(&param_list,
+						AT_CMD_PDP_CONTEXT_READ_DYN_INFO_MTU_INDEX,
+						&(ret_dyn_info->ipv4_mtu));
+			if (ret) {
+				/* Don't care if it fails: */
+				ret = 0;
+				ret_dyn_info->ipv4_mtu = 0;
+			}
+		}
+		if (resp_continues) {
+			at_ptr = next_param_str;
+			iterator++;
+		}
+	}
+
+clean_exit:
+	at_params_list_free(&param_list);
+
+	return ret;
+}
+
+/******************************************************************************/
+
+static void nrf91_non_offload_pdn_event_handler(uint8_t cid, enum pdn_event event, int reason)
+{
+	/* Only default PDN context is supported */
+	if (cid == 0) {
+		if (event == PDN_EVENT_ACTIVATED) {
+			nrf91_non_offload_iface_data.default_pdp_active = true;
+			k_work_schedule(&mdm_socket_work, K_SECONDS(2));
+		} else if (event == PDN_EVENT_DEACTIVATED) {
+			nrf91_non_offload_iface_data.default_pdp_active = false;
+			k_work_schedule(&mdm_socket_work, K_NO_WAIT);
+		}
+	}
+}
+
+/******************************************************************************/
+
+static void nrf91_non_offload_socket_create(void)
+{
+	int ret;
+
+	ret = nrf_socket(NRF_AF_PACKET, NRF_SOCK_RAW, 0);
+	if (ret < 0) {
+		mosh_error("nrf_socket failed %d\n", ret);
+		ret = NO_MDM_SCKT;
+		k_sem_reset(&mdm_socket_sem);
+	} else {
+		k_sem_give(&mdm_socket_sem);
+	}
+	nrf91_non_offload_iface_data.mdm_skct_id = ret;
+}
+
+static void nrf91_non_offload_socket_close(void)
+{
+	k_sem_reset(&mdm_socket_sem);
+	nrf_close(nrf91_non_offload_iface_data.mdm_skct_id);
+	nrf91_non_offload_iface_data.mdm_skct_id = NO_MDM_SCKT;
+}
+
+static void nrf91_non_offload_mdm_socket_worker(struct k_work *unused)
+{
+	ARG_UNUSED(unused);
+	if (nrf91_non_offload_iface_data.default_pdp_active) {
+		if (nrf91_non_offload_iface_data.mdm_skct_id == NO_MDM_SCKT) {
+			char ipv4_addr[NET_IPV4_ADDR_LEN] = { 0 };
+			char ipv6_addr[NET_IPV6_ADDR_LEN] = { 0 };
+			struct sockaddr addr;
+			struct net_if_addr *ifaddr;
+			struct pdn_conn_dyn_params pdn_dyn_info;
+			int len, ret;
+
+			nrf91_non_offload_socket_create();
+			util_get_ip_addr(0, ipv4_addr, ipv6_addr);
+			if (IS_ENABLED(CONFIG_NET_IPV4)) {
+				len = strlen(ipv4_addr);
+				if (len == 0) {
+					mosh_error("Unable to obtain local IPv4 address");
+					return;
+				}
+				ret = net_ipaddr_parse(ipv4_addr, len, &addr);
+				if (!ret) {
+					mosh_error("Unable to parse IPv4 address");
+					return;
+				}
+				ifaddr = net_if_ipv4_addr_add(nrf91_non_offload_iface_data.iface,
+							      &net_sin(&addr)->sin_addr,
+							      NET_ADDR_MANUAL, 0);
+				if (!ifaddr) {
+					mosh_error("Cannot add %s to interface", ipv4_addr);
```

Comment:
Seems that this error is hit when tearing down the default context and setting it back. 
Then IP level connection does not work because IP wasn't set for iface.
I need to work for this still and remove IP addrd from iface when tearing LTE down or something.

Reply:
This was fixed by removing ip addr from net iface when LTE is teared down.